File: Javascript/pattern 4.js
// https://www.codewars.com/kata/55736129f78b30311300010f

function pattern(n){
 return Array.from({length:n}, (_, i) => Array.from({length: n - i}, (_, j) => j + i + 1).join('')).join('\n')
}

const { assert } = require("chai")

describe("Tests", () => {
  it("test", () => {
    assert.strictEqual(pattern(1),"1");
    assert.strictEqual(pattern(2),"12\n2");
    assert.strictEqual(pattern(5),"12345\n2345\n345\n45\n5");
  });
});

⬇️⬇️

/* Example (n = 5):
12345
2345
345
45
5
*/

File: python/pattern 4.py
def pattern(n):
    return '\n'.join(''.join(str(i) for i in range(1 + j, n + 1)) for j in range(n))
        
@test.describe("Basic tests")
def basic():
    @test.it("")
    def f():
        test.assert_equals(pattern(1),"1")
        test.assert_equals(pattern(2),"12\n2")
        test.assert_equals(pattern(5),"12345\n2345\n345\n45\n5")
        test.assert_equals(pattern(0),"")
        test.assert_equals(pattern(-25),"")
        
⬇️⬇️

"""Example:
12345
2345
345
45
5
"""
